vck/src/commonMain/kotlin/at/asitplus/wallet/lib/jws/JwsService.kt

```diff
@@ -193,29 +199,26 @@ class DefaultJwsService(private val cryptoService: CryptoService) : JwsService {
             ?: throw IllegalArgumentException("No encryption in JWE header")
         val epk = header.ephemeralKeyPair
             ?: throw IllegalArgumentException("No epk in JWE header")
-        val z = cryptoService.performKeyAgreement(epk, alg).getOrThrow()
+        val z = cryptoService.performKeyAgreement(epk.toCryptoPublicKey().getOrThrow() as CryptoPublicKey.EC).getOrThrow()
         val intermediateKey = concatKdf(
             z,
             enc,
             header.agreementPartyUInfo,
-            header.agreementPartyVInfo,
-            enc.encryptionKeyLength
+            header.agreementPartyVInfo
         )
-        val key = compositeKey(enc, intermediateKey)
+        require( alg == JweAlgorithm.ECDH_ES )
```

Comment:
But we could do RSA OEAP now?

Reply:
but this thing here would be asymmetric encryption, which signum does not yet support?!

Reply:
the ephemeralkeywithoutcert cannot take a predefined key. but that one was only used for testing, so it is gone now